Lib/pathlib.py

```diff
@@ -347,7 +126,8 @@ def _select_from(self, parent_path, is_dir, exists, scandir):
 class _WildcardSelector(_Selector):
 
     def __init__(self, pat, child_parts, flavour):
-        self.match = flavour.compile_pattern(pat)
+        flags = 0 if flavour is posixpath else re.IGNORECASE
+        self.match = re.compile(fnmatch.translate(pat), flags).fullmatch
```

Comment:
Since the default value is likely never going to change this is fine, but you could also do:
```suggestion
        args = {}
        if flavour is not posixpath:
            args["flags"] = re.IGNORECASE
        self.match = re.compile(fnmatch.translate(pat), **args).fullmatch
```

Reply:
Hm. There's also `re.NOFLAG` as of fea7290a0ec - might be best of both worlds? Will pop a commit in.